backend/lib/routes/info.js

```diff
@@ -58,3 +61,33 @@ async function fetchGardenerVersion () {
     throw err
   }
 }
+
+router.route('/openapi/shoot')
+  .get(async (req, res, next) => {
+    try {
+      const user = req.user
+      const spec = await fetchShootSpec(user)
+      res.send({ spec })
+    } catch (err) {
+      next(err)
+    }
+  })
+
+let shootOpenAPISpecification // Cache, TODO: Need to update cache when apiserver gets updated
+
+async function fetchShootSpec (user) {
+  if (!await canAccessOpenAPI(user)) {
+    throw new Forbidden('User is not allowed to fetch the Open API specification')
+  }
+
+  if (shootOpenAPISpecification) {
+    return shootOpenAPISpecification
+  }
+
+  // Do not use client of user as the result gets cached and returned to other users
+  const res = await dashboardClient.openAPI.get()
+  const spec = await SwaggerParser.dereference(res)
+
+  shootOpenAPISpecification = _.get(spec, ['definitions', 'com.github.gardener.gardener.pkg.apis.core.v1alpha1.Shoot'], {})
```

Comment:
v1alpha1 or v1beta1?

Reply:
has been updated in the meantime, I will switch to v1beta1